### PR TITLE
Fix: Page writing issue

### PR DESCRIPTION
**Pull request by erikjung**
Re: #67 (Question 2) 

I wish I understood a bit more about _why_ this seems to fix the issue, but it does (for my use case anyway).

/CC @mrgerardorodriguez

### CODE REVIEW

**Comment by lyzadanger** [comment]
Hi! I'll pull this branch this morning and take a look at it and see if I can validate the fix _and_ understand what went wrong in the first place :grinning:

**Comment by lyzadanger** [comment]
I need to write some better tests :). Even though you expanded them, they still pass when things are broken because they're kind of testing the wrong thing :P. More soon.

**Comment by lyzadanger** [comment]
@erikjung I fixed the, excuse me, shit out of this. Instead of papering over a convoluted recursion, I defined a more universal way to calculate output paths. Also took care of some housekeeping that the fixes kicked up. Expanded relevant tests and added more fixtures. Would love your feedback as to whether this super duper fixes the whole pile of dumb.

File: src/parse/patterns.js [hunk 1]
```diff
@@ -78,12 +78,11 @@ const hasPatternOrdering = itms => {
...
 function buildPattern (patternObj, options) {
   const patternFile = { path: patternObj.path };
-  const patternId = resourceId(patternFile,
-    options.src.patterns.basedir, 'patterns');
-  checkNamespaceCollision('id', patternObj.data,
-    `Pattern ${patternId}`, options);
+  // const patternId = resourceId(patternFile,
+  //   options.src.patterns.basedir, 'patterns');
+  // checkNamespaceCollision('id', patternObj.data,
+  //   `Pattern ${patternId}`, options);
```
Comment (lyzadanger) [inline src/parse/patterns.js]: Oops. Left some commented-out code.

**Comment by erikjung** [comment]
:+1: 

Things work well now.

**Pull request merged**